src/cpp/src/visual_language/vision_encoder.cpp

```diff
@@ -17,7 +17,11 @@ namespace ov::genai {
 VisionEncoder::VisionEncoder(const std::filesystem::path& model_dir, const std::string& device, const ov::AnyMap properties) {
     auto compiled_model = utils::singleton_core().compile_model(model_dir / "openvino_vision_embeddings_model.xml", device, properties);
     ov::genai::utils::print_compiled_model_properties(compiled_model, "VLM vision embeddings model");
-    m_vision_encoder = compiled_model.create_infer_request();
+    m_ireq_queue_vision_encoder = std::make_unique<CircularBufferQueue<ov::InferRequest>>(
+        compiled_model.get_property(ov::optimal_number_of_infer_requests),
```

Comment:
how many requests will be created here for client PC and Xeon server?

as I see from CPU code, even in case of Xeon we will have 1 infer request (latency mode is default, which means 1 stream per socket)

Reply:
Yes. Tokenizer enables passing properties that would give user possibility to tune this. I can do it similarly for vision encoder and embeddings model.